sdk/dynamicconfig/sdk/metric/controller/push/push.go

```diff
@@ -63,12 +71,24 @@ func New(selector export.AggregatorSelector, exporter export.Exporter, opts ...O
 		c.Timeout = c.Period
 	}
 
+	lock := sync.Mutex{}
+	schedules := []*pb.ConfigResponse_MetricConfig_Schedule{}
+	var instrumentPeriod map[string]pb.ConfigResponse_MetricConfig_Schedule_CollectionPeriod
+
+	var dynamicExtension *sdk.DynamicExtension = nil
+	if c.Notifier != nil {
+		instrumentPeriod = make(map[string]pb.ConfigResponse_MetricConfig_Schedule_CollectionPeriod)
+		dynamicExtension = sdk.NewDynamicExtension(&lock, &schedules, instrumentPeriod)
```

Comment:
I see. So the dynamic extension is sharing the lock with the controlled.

Is there a way to avoid this? Sharing lock complicates the threading model.